lib/commands/index.js

```diff
@@ -27,19 +27,16 @@ module.exports = function(client){
 };
 
 function loadScripts(client, dir) {
-  return fs.readdirAsync(dir).then(function(files){
-    return Promise.all(files.filter(function (file) {
-      return path.extname(file) === '.lua';
-    }).map(function (file) {
-      var longName = path.basename(file, '.lua');
-      var name = longName.split('-')[0];
-      var numberOfKeys = parseInt(longName.split('-')[1]);
+  return fs.readdirAsync(dir).filter(function (file) {
+    return path.extname(file) === '.lua';
+  })
+  .map(function (file) {
+    var longName = path.basename(file, '.lua');
+    var name = longName.split('-')[0];
+    var numberOfKeys = parseInt(longName.split('-')[1]);
 
-      return fs.readFileAsync(path.join(dir, file)).then(function(lua){
-        client.defineCommand(name, { numberOfKeys: numberOfKeys, lua: lua.toString() });
-      }, function(err){
-        console.error('Error reading script file', err);
```

Comment:
The error callback was safely removed because we already attach an error handler to `_this._initializing`.